firewood/src/storage/persisted/mod.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+// TODO: remove this once we have code that uses it
+#![allow(dead_code)]
+
+use bytes::Bytes;
+use std::io::{Error, SeekFrom};
+use std::path::PathBuf;
+use tokio::fs::{File, OpenOptions};
+use tokio::io::{AsyncReadExt, AsyncSeekExt, AsyncWriteExt};
+use tokio::sync::Mutex;
+
+#[derive(Debug)]
+pub struct StoreWrite {
+    offset: u64,
+    data: Bytes,
+}
+
+pub struct PersistedStore {
+    path: PathBuf,
+    fd: Mutex<File>,
+}
+
+impl PersistedStore {
+    pub async fn new(path: PathBuf) -> Result<Self, Error> {
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .truncate(false)
+            .create(true)
+            .mode(0o600)
+            .open(&path)
+            .await?;
+        Ok(Self {
+            path,
+            fd: Mutex::new(file),
+        })
+    }
+
+    // TODO: impl ReadLinearStore when it is async
+    pub async fn stream_from(&self, addr: u64, length: usize) -> Result<Bytes, Error> {
+        let mut file = self.fd.lock().await;
+        file.seek(SeekFrom::Start(addr)).await?;
+        let mut buffer = vec![0; length];
+        file.read_exact(&mut buffer).await?;
+        Ok(Bytes::from(buffer))
+    }
+
+    pub async fn write_all(&mut self, writes: &[StoreWrite]) -> Result<(), Error> {
```

Comment:
How does this hook up to the coordinator, and how do you get notified when all these writes complete?

Reply:
This will be handled by the scheduler PR.

Reply:
Not sure I'm following. Do you want the scheduler to also be async? For the prototype we agreed to keep everything sync, and this async code would get fired off to do the IO.

As it is today, we have no async context so we can't use this method.

I think the input to this is the same structure used by `new` in `Proposed`. Without that, I think we'd have to make a new structure which would involve some copies, unless you have some other idea.